### DISCUSSION TITLE
✅  Help on connection error

### DISCUSSION THREAD

**Discussion by Elfmeterversemmler** [isontheline/pro.webssh.net#425] [Q&A] [answered]
Hi, I just downloaded and installed WebSSH on my iPad and as I have enabled access to my servers only by public key authentication I generated a pair of keys and added the public key to the authroized_keys file of the remote user and copied the private key file to my tablet. Then I added the file by "import from file" to the app and wanted to connect to the remote machine but I am always getting the error "Connection error - unknown private key format".
I have generated the key pair as `ssh-ed25519` as well as `ssh-rsa` with putty where I exported the private key then as OpenSSH key.
I also generated a key pair on the command line with ssh-keygen but that also led to the same error above.
Connection itself is working as I can see messages in the log file on the remote server when not using a key and trying authentication with user/password.
Which format does the key to be in then?
Even enabling logging and setting it to verbose did not shed any light on this.
I also tried to copy and paste the private key by adding it "manually" but even that didn't work - just in case there were some line breaks or so.

**Comment by isontheline** [comment, marked as answer]
Hello @Elfmeterversemmler 👋

PuTTY v3 keys are [not supported](https://github.com/isontheline/pro.webssh.net/issues/381) yet.

The only workaround is to convert your PPK key to PEM :
* https://www.tbs-certificates.co.uk/FAQ/en/putty-ppk-vers-openssl-openssh.html
* https://www.oracle.com/webfolder/technetwork/tutorials/obe/cloud/ggcs/Change_private_key_format_for_Putty/Change_private_key_format_for_Putty.html

Could you give a try?

Have a nice day ☀️

**Reply by Elfmeterversemmler** [reply]
Hi @isontheline,

I had converted the PuTTY keys to OpenSSH keys before I added them to webSSH.
I also created a key pair with ssh-keygen on the command line of a Linux machine but that also didn't work.

Cheers,
Torsten

**Reply by isontheline** [reply]
Hi @Elfmeterversemmler 

It should work : https://github.com/isontheline/pro.webssh.net/discussions/414#discussioncomment-1413327

Please post your errors from /var/log/auth

**Reply by Elfmeterversemmler** [reply]
Hi @isontheline,

I created a key pair as described in #414 before but I probably did not add the public key to the remote server as it seems :thinking: 
I just redid the creation of the key pair and now it works :man_shrugging: 
Anyway, thank you for your assistance and the good app :+1: 

Cheers,
Torsten

**Reply by isontheline** [reply]
Hi @Elfmeterversemmler 👋

Great! Thank you a lot for your feedback 🙏

Have a nice day ☀️
